lib/rex/post/sql/ui/console/interactive_sql_client.rb

```diff
@@ -67,8 +67,19 @@ def _winch
 
   # Try getting multi-line input support provided by Reline, fall back to Readline.
   def _multiline_with_fallback
-    query = _multiline
-    query = _fallback if query[:status] == :fail
+    name = session.type
+    query = {}
+
+    # Multiline (Reline) and fallback (Readline) have separate history contexts as they are two different libraries.
```

Comment:
Just curious, what's the need for both libraries? If one session uses both at different times, would those two histories be merged somewhere/should they be/can they be?

Reply:
> what's the need for both libraries?

Readline is our current library; We're now using the Ruby Reline library (https://github.com/ruby/reline) for multi-line support for SQL queries

In the future we might only have reline - but that's not a blocker for this PR